### PR TITLE
Personalization is rejecting an uncatchable promise

### PR DESCRIPTION
**Pull request by ninaceban**
## Description

We'll resolve the `viewStorageDeferred` promise when the request has failed. When we call the `getView` and the promise has been fulfilled we will just return whatever we have in cache already or `undefined`.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

**Review by jfkhoury — APPROVED** [review]
🏆

**Review by Aaronius — APPROVED** [review]

File: src/components/Personalization/createExecuteCachedViewDecisions.js [hunk 1]
```diff
@@ -19,9 +19,12 @@ export default ({ viewCache, executeViewDecisions, collect }) => {
         executeViewDecisions(viewDecisions);
         return;
       }
-      const xdm = { web: { webPageDetails: { viewName } } };
 
-      collect({ meta: {}, xdm });
+      if (isNonEmptyString(viewName)) {
```
[the remark this answers is not in the document]
Comment (ninaceban) [reply src/components/Personalization/createExecuteCachedViewDecisions.js:23]: if mistakenly the customer would send in `viewName` an empty string we don't want to send a`display` notification

[the remark this answers is not in the document]
Comment (ninaceban) [reply src/components/Personalization/createExecuteCachedViewDecisions.js:23]: I will move it up higher the condition and add a test

**Comment by ninaceban** [comment]
@Aaronius While testing I saw that if we have an empty string passed for the viewName, we'll not render anything, but send a `display` notification. That is why I have added this check. I have moved the check upper, in the createPersonalizationDetails file, where we validate `hasViewName`. I have added a few more tests to cover this. Could you please re-review it? Thanks

**Review by Aaronius — APPROVED** [review]
Looks great. Let's merge!

**Pull request merged**